src/textual/pilot.py

```diff
@@ -107,27 +123,52 @@ async def click(
         message_arguments = _get_mouse_message_arguments(
             target_widget, offset, button=1, shift=shift, meta=meta, control=control
         )
+
+        click_offset = Offset(message_arguments["x"], message_arguments["y"])
+        visible_screen_region = screen.region + screen.scroll_offset
```

Comment:
This *feels* wrong... `click_offset` is relative to the Screen. i.e. (0, 0) is the top left of the screen regardless of the scroll position. No matter how much I scroll (0, 0) will never be out of bounds.

Can you confirm this?

Reply:
The `click_offset` is an offset with respect to the top-left corner of the _virtual_ screen.

So, both `pilot.click(None, (0, 0))` and `pilot.click(Screen, (0, 0))` can be outside of the visible screen if you've scrolled down.

Do you think `pilot.click(None, offset)` should give an offset relative to the visible screen instead?

Reply:
If I do `pilot.click(offset=(0, 0))` I would expect that to mean the top left, regardless of the scroll position. Mouse coordinates come in that way.

Reply:
Ok, but that's _not_ how it works currently.

Do you want me to change it, then?

Reply:
>> Ok, but that's _not_ how it works currently.
> I would be very surprised if that is the case.

I guarantee it is not how it works currently.
<details>
<summary> Test it with this app in `main`. </summary>

```py
import asyncio

from textual.app import App
from textual.widgets import Button, Label


class MyApp(App):
    clicked = False

    def compose(self):
        yield Button()
        for idx in range(200):
            yield Label(str(idx), id=f"lbl{idx}")

    def on_button_pressed(self):
        self.clicked = True


async def main():
    app = MyApp()
    async with app.run_test(size=(80, 24)) as pilot:
        app.query_one("#lbl199").scroll_visible(animate=False)
        await pilot.pause()
        await pilot.click(offset=(0, 0))
        await pilot.pause()

    assert not app.clicked


if __name__ == "__main__":
    asyncio.run(main())

```

</details>

> Clicking (0, 0) should always be possible. Clicking (0, -1) should always be out of bounds.

I'll make it happen, then. `offset` with `selector=None` will be relative to visible screen. If any selector is provided, then the offset is relative to the top-left corner of the selected widget.

Reply:
Something wrong is not right. I think the app above proves your point, not mine.
I'll figure out what I'm missing and I'll find out why the tests didn't catch it.

Reply:
For the record, what I was missing was the correct interpretation of the screen coordinate system.
The `Screen` doesn't move and its top-left corner always matches the top-left corner of the terminal.